web/src/components/TimeSlider.tsx

```diff
@@ -17,18 +85,61 @@ function TimeSlider({ onChange, numberOfEntries, selectedIndex }: TimeSliderProp
       aria-label="value"
       className="relative mb-2 flex h-5 w-full touch-none items-center"
     >
-      <SliderPrimitive.Track className="relative h-2.5 w-full grow rounded-sm bg-gray-100 dark:bg-gray-700">
+      <SliderPrimitive.Track
+        className="relative h-2.5 w-full grow rounded-sm"
+        style={{ background: trackBackground }}
+      >
         <SliderPrimitive.Range />
       </SliderPrimitive.Track>
       <SliderPrimitive.Thumb
         data-test-id="time-slider-input"
-        className={`block h-6 w-6 rounded-full bg-white bg-[url("/images/slider-thumb.svg")]
+        className={`block h-6 w-6 rounded-full bg-white
           bg-center bg-no-repeat shadow-3xl focus:outline-none
           focus-visible:ring focus-visible:ring-brand-green/10
           focus-visible:ring-opacity-75 dark:bg-gray-400 dark:focus-visible:ring-white/70`}
+        style={{ backgroundImage: `url(/images/${thumbIcon})` }}
       ></SliderPrimitive.Thumb>
     </SliderPrimitive.Root>
   );
 }
 
+export function TimeSliderWithoutNight(props: TimeSliderProps) {
+  const isDarkModeEnabled = useDarkMode();
+  const thumbIcon = getThumbIcon();
+  const trackBackground = getTrackBackground(isDarkModeEnabled, props.numberOfEntries);
+  return (
+    <TimeSliderBasic {...props} trackBackground={trackBackground} thumbIcon={thumbIcon} />
+  );
+}
+
+export function TimeSliderWithNight(props: TimeSliderProps) {
+  const nightTimeSets = useNightTimes();
+  const isDarkModeEnabled = useDarkMode();
+
+  const thumbIcon = getThumbIcon(props.selectedIndex || 0, nightTimeSets);
+  const trackBackground = getTrackBackground(
+    isDarkModeEnabled,
+    props.numberOfEntries,
+    nightTimeSets
+  );
+
+  return (
+    <TimeSliderBasic {...props} trackBackground={trackBackground} thumbIcon={thumbIcon} />
+  );
+}
+
+function TimeSlider(props: TimeSliderProps) {
+  const zoneId = getZoneFromPath();
+  const [timeAverage] = useAtom(timeAverageAtom);
+
+  const showNightTime =
+    zoneId && timeAverage === TimeAverages.HOURLY && props.numberOfEntries === 24;
```

Comment:
I believe this should be 25?
As we show from 09:00 to 09:00.

Is the props length even needed though?

Reply:
Apparently not, logging `props.numberOfEntries` returns 24 when in hourly :)

And yes, if we remove it there's an annoying flickering going when switching between time aggregates due to the way things are rendered:

https://github.com/electricitymaps/electricitymaps-contrib/assets/3296643/b8185625-a245-4747-ab20-0eb8e7e75288

Reply:
Oh that was a nasty bug. Let's keep it in there!